### ISSUE TITLE
Illegal memory access in gpu::octree

### ISSUE THREAD

**Issue by SergioRAgostinho** [PointCloudLibrary/pcl#2103]
I detected an illegal memory access, in `octree_builder.cu:246`. I'm not sure how reproductible this is.

- The [pcd file](https://github.com/PointCloudLibrary/pcl/files/1503537/pointcloud.txt)  (just replace the extension if needed).
- No error is triggered in my system if the point cloud as 950 points instead of 951. 
- Hardware: GeForce GTX 1060
- Driver: 384.90
- CUDA Version:  8.0

```cpp
#include <pcl/gpu/octree/octree.hpp>
#include <pcl/io/pcd_io.h>

int main (const int argc, const char** const argv)
{
  pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZ>);
	pcl::io::loadPCDFile(argv[1], *cloud);

  std::cout << "Point Cloud Info:\n"
            << "height: " << cloud->height
            << " width: " << cloud->width
            << std::endl;
	pcl::gpu::Octree::PointCloud cloud_device;
	cloud_device.upload(cloud->points);
	pcl::gpu::Octree octree_device;
	octree_device.setCloud(cloud_device);
	octree_device.build();

  return 0;
}
```

```cmake
cmake_minimum_required(VERSION 3.0.0 FATAL_ERROR)
project(gpu_octree VERSION 0.1.0 LANGUAGES CXX)

find_package(PCL 1.8.1.99 REQUIRED)

add_executable (${PROJECT_NAME} "${PROJECT_NAME}.cpp")
target_include_directories(${PROJECT_NAME} PUBLIC ${PCL_INCLUDE_DIRS})
target_link_libraries (${PROJECT_NAME} ${PCL_LIBRARIES})
```

**Comment by atenpas** [comment]
I've encountered a similar _illegal memory access_ in `normal_3d.cu:251`. This happens about every 2nd call to `computeNormals`.

- Hardware: 2 x GeForce GTX 1080
- PCL: 1.9.1
- NVIDIA driver: 390.87
- CUDA: 9.1

**Comment by b4l8** [comment]
same issue with normalestimation, in octree_builder.cu:403

    Hardware: GeForce GTX 1070
    PCL: 1.9.0
    NVIDIA driver: 384.130
    CUDA: 9.1

**Comment by phil0stine** [comment]
Same issue with normal estimation, 
Hardware: Quadro K3100M
PCL: 1.9.1
NVIDIA driver: 390.116
CUDA: 9.0

**Comment by SergioRAgostinho** [comment]
Hopefully closed with the merge of #3627
